libs/db/MapDB.ts

```diff
@@ -0,0 +1,95 @@
+import admin from 'firebase-admin';
+import type {
+  CollectionReference,
+  Firestore,
+  WriteResult,
+} from 'firebase-admin/firestore';
+import type { DecodedIdToken } from 'firebase-admin/auth';
+import IMapData from '../../interfaces/IMapDB';
+import type { MapData } from '../../types/types';
+class MapDB implements IMapData {
+  private db: Firestore;
+
+  constructor() {
+    this.db = admin.firestore();
+  }
+
+  // private initializeFirebase(): App {
+  //   try {
+  //     return admin.initializeApp({
+  //       credential: admin.credential.cert(firebaseConfig as ServiceAccount),
+  //     });
+  //   } catch (e: any) {
+  //     throw new Error(e);
+  //   }
+  // }
+
+  public async set(data: MapData): Promise<WriteResult> {
+    try {
+      const collectionRef: CollectionReference = this.db.collection('map');
+      return await collectionRef.doc().set(data);
+    } catch (e: any) {
+      throw new Error(e);
+    }
+  }
+  public async getByID(id: string): Promise<MapData | null> {
+    try {
+      const collectionRef: CollectionReference = this.db.collection('map');
+      const docRef = await collectionRef.doc(id).get();
+      if (docRef.exists) {
+        return docRef.data() as MapData;
+      }
+      return null;
+    } catch (e: any) {
+      throw new Error(e);
+    }
+  }
+  public async getByCoordinates(coordinates: {
+    long: number;
+    lang: number;
+  }): Promise<MapData | number | null> {
```

Comment:
If you are inly returning the size, then only set the return type to number and not MapData and null it will and should always only be number